sage-ios/SAGE/SAGE/RequestHoursView.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  RequestHoursView.swift
+//  SAGE
+//
+//  Created by Andrew on 10/27/15.
+//  Copyright © 2015 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+import BSKeyboardControls
+
+class RequestHoursView: UIView {
+    
+    var dateField =  FormFieldItem()
+    var startTimeField = FormFieldItem()
+    var endTimeField = FormFieldItem()
+    var commentField = FormTextItem()
+    private var keyboardControls = BSKeyboardControls()
+    private var scrollView = UIScrollView()
+    private var dateFormatter = NSDateFormatter()
+
+    //
+    // MARK: - Initialization
+    //
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.backgroundColor = UIColor.whiteColor()
+        self.setupSubviews()
+        self.keyboardControls.fields = [
+            self.dateField.textField,
+            self.startTimeField.textField,
+            self.endTimeField.textField,
+            self.commentField.textView
+        ]
+        self.keyboardControls.barTintColor = UIColor.mainColor
+        self.keyboardControls.delegate = self
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func layoutSubviews() {
+        self.scrollView.fillWidth()
+        self.scrollView.fillHeight()
+        
+        self.dateField.fillWidth()
+        
+        self.startTimeField.setY(CGRectGetMaxY(self.dateField.frame))
+        self.startTimeField.fillWidth()
+        
+        self.endTimeField.setY(CGRectGetMaxY(self.startTimeField.frame))
+        self.endTimeField.fillWidth()
+
+        self.commentField.setY(CGRectGetMaxY(self.endTimeField.frame))
+        self.commentField.fillWidth()
+        
+        self.scrollView.contentSize = CGSizeMake(CGRectGetWidth(self.frame), CGRectGetMaxY(self.commentField.frame))
+    }
+    
+    //
+    // MARK: - Event handlers
+    //
+    @objc private func datePicked(sender: UIDatePicker!) {
```

Comment:
do selector/actions need to be objective-c tagged? I didn't tag mine and they work